users/tests/test_views.py

```diff
@@ -156,3 +156,56 @@ def test_user_update_view_should_not_update_user_on_post_if_form_is_invalid(self
         self.user.refresh_from_db()
         self.assertEqual(response.status_code, 200)
         self.assertEqual(phone_number_before_request, self.user.phone_number)
+
+
+class UserUpdateByAdminTests(TestCase):
+    def setUp(self):
+        self.user = CustomUser(
+            email="testuser@codepoets.it", password="newuserpasswd", first_name="John", last_name="Doe", country="PL"
+        )
+        self.user.full_clean()
+        self.user.save()
+        self.user_admin = AdminUserFactory()
+        self.client.force_login(self.user_admin)
+        self.user.refresh_from_db()
+
+    def test_user_update_by_admin_view_should_display_user_details_on_get(self):
+        response = self.client.get(path=reverse("custom-user-update-by-admin", args=(self.user.pk,)))
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, self.user.email)
+        self.assertContains(response, self.user.first_name)
+        self.assertContains(response, self.user.last_name)
+
+    def test_user_update_by_admin_view_should_not_render_non_existing_user(self):
+        not_existing_pk = 1000
+        response = self.client.get(path=reverse("custom-user-update-by-admin", args=(not_existing_pk,)))
+        self.assertEqual(response.status_code, 404)
+
+    def test_user_update_by_admin_view_should_update_user_on_post(self):
+        new_phone_number = generate_random_phone_number(constants.PHONE_NUMBER_MIN_LENGTH)
+        response = self.client.post(
+            path=reverse("custom-user-update-by-admin", args=(self.user.pk,)),
+            data={
+                "email": self.user.email,
+                "phone_number": new_phone_number,
+                "user_type": CustomUser.UserType.EMPLOYEE.name,
+            },
+        )
+        self.user.refresh_from_db()
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(new_phone_number, self.user.phone_number)
+
+    def test_user_update_by_admin_view_should_not_update_user_on_post_if_form_is_invalid(self):
+        phone_number_before_request = self.user.phone_number
+        invalid_phone_number = generate_random_phone_number(constants.PHONE_NUMBER_MIN_LENGTH - 1)
+        response = self.client.post(
+            path=reverse("custom-user-update-by-admin", args=(self.user.pk,)),
+            data={
```

Comment:
Looks like this `data` could be extracted and defined in `setUp` ie.

Reply:
`data` is not exactly the same, so it cannot be a field, but I've made a small helper method for it :)